### PR TITLE
feat: add Apple M1 (osx-arm64) support

### PR DESCRIPTION
**Pull request by jfcherng**
Resolves https://github.com/sublimelsp/LSP-lua/issues/27
Resolves https://discord.com/channels/280102180189634562/645268178397560865/927245849757646918

---

Someone has to add `osx-arm64` to the `platforms` in https://github.com/sublimelsp/repository/blob/d54c4fd3aa2ca23e5e7803c538ed5921e5674e0f/repository.json#L447-L460? It seems to be auto generated somehow.

### CODE REVIEW

**Comment by rchl** [comment]
Are you sure that `osx-arm64` is supported by PC? I seem to recall that it wasn't.

File: plugin.py [hunk 1]
```diff
@@ -35,16 +33,17 @@ def zipfile(cls) -> str:
...
         return {
-            "linux": "Linux",
-            "windows": "Windows",
-            "osx": "macOS"
-        }[sublime.platform()]
+            "linux_x64": "linux-x64",
+            "osx_arm64": "darwin-arm64",
+            "osx_x64": "darwin-x64",
+            "windows_x64": "win32-x64",
+        }[sublime.executable_hash()[1]]
```
Comment (rchl) [inline plugin.py:42]: Maybe instead of using undocumented API do "sublime.platform() - sublime.arch()" instead?

**Comment by jfcherng** [comment]
> Are you sure that `osx-arm64` is supported by PC? I seem to recall that it wasn't.

There is another solution now. We now support all ST 4 platform/arch. So we can remove the `platforms` constraint.

**Comment by rchl** [comment]
> Though at the same time, no one gives comments on [wbond/package_control_channel#8361](https://github.com/wbond/package_control_channel/pull/8361)

I suppose I was thinking about that.

**Review by rchl — APPROVED** [review]

**Pull request merged**